### ISSUE TITLE
Can't get my startup code to be at the start of the binary

### ISSUE THREAD

**Issue by dominicbeesley** [hth313/Calypsi-tool-chains#45]
Hi,

I've been trying out the Calypsi 65816 tools today and I've managed to get a minimal working program that pokes direct to memory. However, I don't seem to be able to make the linker output in binary mode in the order I'd like. I would like my entry code (in enter.s) to be at the very start of the binary in the small attached example. Whatever I do (order of objects sent to link; --cstartup=mine; placing entry code in own section) it always seems to place the main function at the start of the binary file and my enter.s code somewhere later on.

I realise I'm probably doing something dumb in the .scm file.

Thanks

Dom
[test816.zip](https://github.com/user-attachments/files/17232924/test816.zip)

**Comment by hth313** [comment]
In order to place a section at a specific address you need to give it a unique name. In `enter.s` I renamed the section to `rootcode`:
```
	         .rtmodel cstartup,"mine"
		.extern main
		
		.section rootcode, root, noreorder
```
Then you need place that in the `.scm` file. There are currently some issues with placement groups and fixed addressing that I am about to look into, so as a work around I suggest you create two separate memories, one for read-only things and one for data:
```
(define memories
	'(
	(memory directPage
		(address (#x0 . #xff))
		(section registers zzpage zpage ztiny)
	)
	(memory RAM1-code (address (#x0E00 . #x1fff))
		(section (rootcode #x0E00) startup code data_init_table))
	(memory RAM1-data (address (#x2000 . #x4fff))
		(section stack heap data zdata))
	(block stack (size #x800)) ; machine stack size
	(block heap  (size #x800)) ; heap size
	(base-address _DirectPageStart directPage 0)
))
```

**Comment by dominicbeesley** [comment]
Ah, thanks, I think I'd tried that but using placement groups. It would be nice to be able to instruct the linker to just place the data after the read only stuff to make as small a binary as possible. 

Another feature would be to have data_init_table be able to overlay the other sections as it is only required at startup.

thanks for your help

D

**Comment by hth313** [comment]
Fixed in release 5.7
